### PR TITLE
replace np.flip with indexing [::-1]

### LINKED ISSUE #262: [BUG] Can't import latest version of xgcm
**What happened**:
I installed `xgcm` from commit b2fc97e and couldn't import it.
**What you expected to happen**:
I expected to be able to import.
**Minimal Complete Verifiable Example**:

```bash
python -m pip install --user git+https://github.com/xgcm/xgcm.git@b2fc97e29c6e262f69dc699d4c31dd2ad3ee53fc --upgrade

  Cloning https://github.com/xgcm/xgcm.git (to revision b2fc97e29c6e262f69dc699d4c31dd2ad3ee53fc) to /scratch/v45/nc3020/tmp/pip-req-build-94nh9e6b
  Running command git clone -q https://github.com/xgcm/xgcm.git /scratch/v45/nc3020/tmp/pip-req-build-94nh9e6b
Requirement already satisfied, skipping upgrade: xarray>=0.14.1 in /g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages (from xgcm==0.5.0+5.gb2fc97e) (0.16.1)
Requirement already satisfied, skipping upgrade: dask in /g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages (from xgcm==0.5.0+5.gb2fc97e) (2.28.0)
Requirement already satisfied, skipping upgrade: numpy in ./.local/lib/python3.7/site-packages (from xgcm==0.5.0+5.gb2fc97e) (1.19.1)
Requirement already satisfied, skipping upgrade: future in ./.local/lib/python3.7/site-packages (from xgcm==0.5.0+5.gb2fc97e) (0.18.2)
Requirement already satisfied, skipping upgrade: docrep<=0.2.7 in ./.local/lib/python3.7/site-packages (from xgcm==0.5.0+5.gb2fc97e) (0.2.7)
Requirement already satisfied, skipping upgrade: setuptools>=38.4 in /g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages (from xarray>=0.14.1->xgcm==0.5.0+5.gb2fc97e) (49.2.0.post20200712)
Requirement already satisfied, skipping upgrade: pandas>=0.25 in /g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages (from xarray>=0.14.1->xgcm==0.5.0+5.gb2fc97e) (1.1.2)
Requirement already satisfied, skipping upgrade: pyyaml in /g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages (from dask->xgcm==0.5.0+5.gb2fc97e) (5.3.1)
Requirement already satisfied, skipping upgrade: six in /g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages (from docrep<=0.2.7->xgcm==0.5.0+5.gb2fc97e) (1.15.0)
Requirement already satisfied, skipping upgrade: pytz>=2017.2 in /g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages (from pandas>=0.25->xarray>=0.14.1->xgcm==0.5.0+5.gb2fc97e) (2020.1)
Requirement already satisfied, skipping upgrade: python-dateutil>=2.7.3 in /g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages (from pandas>=0.25->xarray>=0.14.1->xgcm==0.5.0+5.gb2fc97e) (2.8.1)
Building wheels for collected packages: xgcm
  Building wheel for xgcm (setup.py) ... done
  Created wheel for xgcm: filename=xgcm-0.5.0+5.gb2fc97e-py3-none-any.whl size=60190 sha256=4901447ef2bb3b5bfae8f7c6ed399f70223a461406d9d33b0c4d4e72fafa9973
  Stored in directory: /home/552/nc3020/.cache/pip/wheels/31/e5/2a/2ebc2c994ecc59d2a757e2f93a388cddba68c78c374254da23
Successfully built xgcm
Installing collected packages: xgcm
Successfully installed xgcm-0.5.0+5.gb2fc97e
```
```python
Python 3.7.8 | packaged by conda-forge | (default, Jul 31 2020, 02:25:08)
[GCC 7.5.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import xgcm
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/552/nc3020/.local/lib/python3.7/site-packages/xgcm/__init__.py", line 6, in <module>
    from .grid import Grid, Axis
  File "/home/552/nc3020/.local/lib/python3.7/site-packages/xgcm/grid.py", line 19, in <module>
    from .transform import linear_interpolation, conservative_interpolation
  File "/home/552/nc3020/.local/lib/python3.7/site-packages/xgcm/transform.py", line 18, in <module>
    nopython=True,
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/npyufunc/decorators.py", line 179, in wrap
    guvec.add(fty)
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/npyufunc/ufuncbuilder.py", line 215, in add
    self.nb_func, targetoptions, sig)
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/npyufunc/ufuncbuilder.py", line 146, in _compile_element_wise_function
    cres = nb_func.compile(sig, **targetoptions)
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/npyufunc/ufuncbuilder.py", line 95, in compile
    return self._compile_core(sig, flags, locals)
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/npyufunc/ufuncbuilder.py", line 130, in _compile_core
    flags=flags, locals=locals)
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/compiler.py", line 551, in compile_extra
    return pipeline.compile_extra(func)
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/compiler.py", line 331, in compile_extra
    return self._compile_bytecode()
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/compiler.py", line 393, in _compile_bytecode
    return self._compile_core()
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/compiler.py", line 373, in _compile_core
    raise e
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/compiler.py", line 364, in _compile_core
    pm.run(self.state)
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/compiler_machinery.py", line 347, in run
    raise patched_exception
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/compiler_machinery.py", line 338, in run
    self._runPass(idx, pass_inst, state)
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/compiler_lock.py", line 32, in _acquire_compile_lock
    return func(*args, **kwargs)
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/compiler_machinery.py", line 302, in _runPass
    mutated |= check(pss.run_pass, internal_state)
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/compiler_machinery.py", line 275, in check
    mangled = func(compiler_state)
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/typed_passes.py", line 95, in run_pass
    raise_errors=self._raise_errors)
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/typed_passes.py", line 67, in type_inference_stage
    infer.propagate(raise_errors=raise_errors)
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/typeinfer.py", line 985, in propagate
    raise errors[0]
numba.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Use of unsupported NumPy function 'numpy.flip' or unsupported use of the function.

File ".local/lib/python3.7/site-packages/xgcm/transform.py", line 23:
def _interp_1d_linear(phi, theta, target_theta_levels, mask_edges, output):
    <source elided>
    if theta[-1] < theta[0]:
        theta = np.flip(theta)
        ^

[1] During: typing of get attribute at /home/552/nc3020/.local/lib/python3.7/site-packages/xgcm/transform.py (23)

File ".local/lib/python3.7/site-packages/xgcm/transform.py", line 23:
def _interp_1d_linear(phi, theta, target_theta_levels, mask_edges, output):
    <source elided>
    if theta[-1] < theta[0]:
        theta = np.flip(theta)
        ^
```

**Anything else we need to know?**:

**xgcm version and environment**:
xgcm 0.5.0+5.gb2fc97e
<details>

</details>

### PR DESCRIPTION
**Pull request by navidcy**
- [X] Closes #262 

This is an attempt to fix #262. I simply did what @rabernat suggested. With this change, when I installed `xgcm` from my fork this seem OK.

```python
(base) [nc3020@gadi-login-02 ~]$ python -m pip install --user git+https://github.com/navidcy/xgcm.git@fa8fb6d9ed59201496855e3a303506263e09a007 --upgrade
Collecting git+https://github.com/navidcy/xgcm.git@fa8fb6d9ed59201496855e3a303506263e09a007
  Cloning https://github.com/navidcy/xgcm.git (to revision fa8fb6d9ed59201496855e3a303506263e09a007) to /scratch/v45/nc3020/tmp/pip-req-build-m8f_y12m
  Running command git clone -q https://github.com/navidcy/xgcm.git /scratch/v45/nc3020/tmp/pip-req-build-m8f_y12m
  Running command git checkout -q fa8fb6d9ed59201496855e3a303506263e09a007
Requirement already satisfied, skipping upgrade: xarray>=0.14.1 in /g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages (from xgcm==0.3.0+79.gfa8fb6d) (0.16.1)
Requirement already satisfied, skipping upgrade: dask in /g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages (from xgcm==0.3.0+79.gfa8fb6d) (2.28.0)
Requirement already satisfied, skipping upgrade: numpy in ./.local/lib/python3.7/site-packages (from xgcm==0.3.0+79.gfa8fb6d) (1.19.1)
Requirement already satisfied, skipping upgrade: future in ./.local/lib/python3.7/site-packages (from xgcm==0.3.0+79.gfa8fb6d) (0.18.2)
Requirement already satisfied, skipping upgrade: docrep<=0.2.7 in ./.local/lib/python3.7/site-packages (from xgcm==0.3.0+79.gfa8fb6d) (0.2.7)
Requirement already satisfied, skipping upgrade: setuptools>=38.4 in /g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages (from xarray>=0.14.1->xgcm==0.3.0+79.gfa8fb6d) (49.2.0.post20200712)
Requirement already satisfied, skipping upgrade: pandas>=0.25 in /g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages (from xarray>=0.14.1->xgcm==0.3.0+79.gfa8fb6d) (1.1.2)
Requirement already satisfied, skipping upgrade: pyyaml in /g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages (from dask->xgcm==0.3.0+79.gfa8fb6d) (5.3.1)
Requirement already satisfied, skipping upgrade: six in /g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages (from docrep<=0.2.7->xgcm==0.3.0+79.gfa8fb6d) (1.15.0)
Requirement already satisfied, skipping upgrade: pytz>=2017.2 in /g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages (from pandas>=0.25->xarray>=0.14.1->xgcm==0.3.0+79.gfa8fb6d) (2020.1)
Requirement already satisfied, skipping upgrade: python-dateutil>=2.7.3 in /g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages (from pandas>=0.25->xarray>=0.14.1->xgcm==0.3.0+79.gfa8fb6d) (2.8.1)
Building wheels for collected packages: xgcm
  Building wheel for xgcm (setup.py) ... done
  Created wheel for xgcm: filename=xgcm-0.3.0+79.gfa8fb6d-py3-none-any.whl size=60195 sha256=5dd6c3bec21b5470619f5a4add86b771bb638d48482f74074bdef60d2d1c9c6d
  Stored in directory: /home/552/nc3020/.cache/pip/wheels/1f/c4/82/1b4b177ce7ab3fe0b1aaf0c11bf0e0b3d5755d1ebeafc80d62
Successfully built xgcm
Installing collected packages: xgcm
  Attempting uninstall: xgcm
    Found existing installation: xgcm 0.5.0
    Uninstalling xgcm-0.5.0:
      Successfully uninstalled xgcm-0.5.0
Successfully installed xgcm-0.3.0+79.gfa8fb6d
(base) [nc3020@gadi-login-02 ~]$ python
Python 3.7.8 | packaged by conda-forge | (default, Jul 31 2020, 02:25:08)
[GCC 7.5.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import xgcm
>>>
```

@jbusecke could you have a look? I'm not sure whether or what test could have caught this.

### CODE REVIEW

**Comment by navidcy** [comment]
I did notice that when I installed from my latest commit I see
```
Successfully installed xgcm-0.3.0+79.gfa8fb6d
```
Why `0.3.0+...`?

**Comment by jbusecke** [comment]
Great! Thanks for the contribution. The modifications look good. I think we should make sure that this all is tested with an older version in the ci. Id suggest you pin an older version of number e.g. [here](https://github.com/navidcy/xgcm/blob/fa8fb6d9ed59201496855e3a303506263e09a007/ci/environment-py3.6.yml#L7). Id suggest `numba<=0.48`, unless somebody sees a reason to go even lower? That way we make sure that all the tests are executed at least once with the old numba version. 
Additionally, you should remove the version restriction and modify the warning text [here](https://github.com/navidcy/xgcm/blob/fa8fb6d9ed59201496855e3a303506263e09a007/xgcm/grid.py#L853-L855). 
Also of course, add this to [whats-new.rst](https://github.com/navidcy/xgcm/blob/master/doc/whats-new.rst) and get the credit for improving xgcm!

That should be all IMO.

Now to your installation issue, I dont know why its defaulting to that old version...I think there was a similar issue in #262, and a fresh environment did the trick. As long as things go through CI I think we are good. Does anyone else have a clue why conda would do that?

**Comment by navidcy** [comment]
pin `numba<=0.48` in all python version envs, right?

**Comment by navidcy** [comment]
@jbusecke, feel free to edit at your liking

**Comment by navidcy** [comment]
hm seems like test fail, I guess since #205 requires `numba 0.49`?

**Comment by jbusecke** [comment]
I changed the numba version back to the latest one for the python 3.7 and 3.8 environment, since we only need one case where we test against the outdated version. In the more recent ci environments we want to test against the latest and greatest.

I think the test will pass once you release the version restriction in `grid.py` [here](https://github.com/navidcy/xgcm/blob/fa8fb6d9ed59201496855e3a303506263e09a007/xgcm/grid.py#L853-L855).

I would change it to ( I tried to edit it but gh was being weird):
```
# check optional numba dependency
        if numba is None":
            raise ImportError(
                "The transform functionality of xgcm requires numba. Install using `conda install numba` "
            )
```

File: doc/whats-new.rst [hunk 1]
```diff
@@ -3,6 +3,15 @@
 What's New
 ===========
 
+v0.6.0 (unreleased)
+-------------------------
+
+Bug fixes
+~~~~~~~~~
+- Fix numba related bug replacing :py:meth:`np.flip` with indexing :py:meth:`[::-1]` (:pull:`263`, :issue:`262`). By `Navid Constantinou <https://github.com/navidcy>`_.
```
Comment (jbusecke) [inline doc/whats-new.rst:11]: Just one nitpick: I would prefer a more general description like: "Adding support for older numba versions (<0.49)"

**Review by jbusecke — APPROVED** [review]
This looks good to me (pending tiny change in the docs). What do others think?

File: doc/whats-new.rst (hunk 1 above)
Comment (navidcy) [reply doc/whats-new.rst:11]: Your suggestion is actually much better.

**Comment by rabernat** [comment]
Fantastic, thanks @navidcy!

**Pull request merged**